xds/src/test/java/io/grpc/xds/XdsLoadReportStoreTest.java

```diff
@@ -0,0 +1,261 @@
+    List<UpstreamLocalityStats> upstreamLocalityStatsList = new ArrayList<>();
+    upstreamLocalityStatsList.add(buildUpstreamLocalityStats(locality1, 0, 0, 0));
+    upstreamLocalityStatsList.add(buildUpstreamLocalityStats(locality2, 0, 0, 1));
+    expectedLoadReport = buildClusterStats(interval, upstreamLocalityStatsList);
+    assertClusterStatsEqual(expectedLoadReport, loadStore.generateLoadReport(interval));
+  }
+
+  @Test
+  public void testLocalityRemovedContinueRecordingOngoingLoads() {
+    loadStore.addLocality(TEST_LOCALITY);
+    assertThat(localityLoadCounters).containsKey(TEST_LOCALITY);
+    PickResult pickResult = PickResult.withSubchannel(fakeSubchannel);
+    PickResult interceptedPickResult = loadStore.interceptPickResult(pickResult, TEST_LOCALITY);
+    ClientStreamTracer tracer = interceptedPickResult
+        .getStreamTracerFactory()
+        .newClientStreamTracer(STREAM_INFO, new Metadata());
+
+    Duration interval = Duration.newBuilder().setNanos(342).build();
+    ClusterStats expectedLoadReport = buildClusterStats(interval,
+        Collections.singletonList(buildUpstreamLocalityStats(TEST_LOCALITY, 0, 0, 1)));
+    assertClusterStatsEqual(expectedLoadReport, loadStore.generateLoadReport(interval));
+    // Remote balancer instructs to remove the locality while client has in-progress calls
+    // to backends in the locality, the XdsClientLoadStore continues tracking its load stats.
+    loadStore.removeLocality(TEST_LOCALITY);
+    assertThat(localityLoadCounters).containsKey(TEST_LOCALITY);
+    expectedLoadReport = buildClusterStats(interval,
+        Collections.singletonList(buildUpstreamLocalityStats(TEST_LOCALITY, 0, 0, 1)));
+    assertClusterStatsEqual(expectedLoadReport, loadStore.generateLoadReport(interval));
+
+    tracer.streamClosed(Status.OK);
+    expectedLoadReport = buildClusterStats(interval,
+        Collections.singletonList(buildUpstreamLocalityStats(TEST_LOCALITY, 1, 0, 0)));
+    assertClusterStatsEqual(expectedLoadReport, loadStore.generateLoadReport(interval));
+    assertThat(localityLoadCounters).doesNotContainKey(TEST_LOCALITY);
+  }
+
+  private UpstreamLocalityStats buildUpstreamLocalityStats(Locality locality, long callsSucceed,
+      long callsFailed, long callsInProgress) {
+    return UpstreamLocalityStats.newBuilder()
+        .setLocality(locality)
+        .setTotalSuccessfulRequests(callsSucceed)
+        .setTotalErrorRequests(callsFailed)
+        .setTotalRequestsInProgress(callsInProgress)
+        .build();
+  }
+
+  private ClusterStats buildClusterStats(Duration interval,
+      List<UpstreamLocalityStats> upstreamLocalityStatsList) {
+    return ClusterStats.newBuilder().setClusterName(SERVICE_NAME)
+        .addAllUpstreamLocalityStats(upstreamLocalityStatsList)
+        .setLoadReportInterval(interval)
+        .build();
+  }
+
+  private void assertClusterStatsEqual(ClusterStats stats1, ClusterStats stats2) {
```

Comment:
Why can't you use the equality of `ClusterStats`?

Reply:
`ClusterStats` contains a list of `UpstreamLocalityStats`, for which the ordering we do not know. Two lists of `UpstreamLocalityStats` containing same instances should be considered "equal".